src/orion/core/io/experiment_builder.py

```diff
@@ -505,10 +505,16 @@ def _instantiate_algo(space, max_trials, config=None, ignore_unavailable=False):
     try:
         backported_config = backward.port_algo_config(config)
         algo_constructor = algo_factory.get_class(backported_config.pop("of_type"))
-        algo = SpaceTransformAlgoWrapper(
-            algo_constructor, space=space, **backported_config
-        )
-        algo.algorithm.max_trials = max_trials
+
+        requirements = backward.get_algo_requirements(algo_constructor)
+        from orion.core.worker.transformer import build_required_space
+
+        original_space = space
+        transformed_space = build_required_space(space, **requirements)
```

Comment:
There should be a helper function to this. Building algos outside experiment creation will be cumbersome otherwise since algos can hardly be used on non transformed spaces based on their requirements.

Reply:
Where should this helper function be placed?

Reply:
Added a helper function in 0ae1340407e9e37909ecb66d2516dc325ec9c6f7, let me know where you'd like it to be moved.

Reply:
I think inside the wrapper module would make sense.

Reply:
Fixed in 52ba2515be09d9ebd5e500dfb10b28a34851f978